invoice/services/bill.py

```diff
@@ -0,0 +1,59 @@
+from typing import Dict, Union, List
+
+from invoice.models import Bill, BillItem
+from invoice.services.base import BaseService
+from invoice.services.service_utils import _get_generic_type
+from invoice.validation.bill import BillModelValidation, BillItemStatus
+from core.signals import *
+
+
+class BillService(BaseService):
+
+    OBJECT_TYPE = Bill
+
+    def __init__(self, user, validation_class: BillModelValidation = BillModelValidation):
+        super().__init__(user, validation_class)
+        self.validation_class = validation_class
+
+    def _base_payload_adjust(self, bill_data):
+        return self._evaluate_generic_types(bill_data)
+
+    def bill_validate_items(self, bill: Bill):
+        # TODO: Implement after calculation rules available
+        pass
+
+    def bill_match_items(self, bill: Bill) \
+            -> Dict[str, Union[BillItemStatus, Dict[BillItem, List[BillItemStatus]]]]:
```

Comment:
It's just a note, but `Union` is for X | Y.  In docstring you're more specific; we're always returning ''subject' with BillItemStatus and 'line_items' with bill items and statuses. It'd might be better idea to use `TypedDict` for this hint so it'll match docstring information.

Reply:
Ok. I will check this. Thx for this note.